tests/functional/aws-node-sdk/test/bucket/getWebsite.js

```diff
@@ -0,0 +1,94 @@
+import assert from 'assert';
+
+import withV4 from '../support/withV4';
+import BucketUtility from '../../lib/utility/bucket-util';
+import { WebsiteConfigTester } from '../../lib/utility/website-util';
+
+const bucketName = 'testgetwebsitebucket';
+const ruleRedirect1 = {
+    HostName: 'test',
+    ReplaceKeyPrefixWith: 'documents/',
+};
+const ruleCondition1 = {
+    KeyPrefixEquals: 'docs/',
+};
+const ruleRedirect2 = {
+    HttpRedirectCode: 302,
+};
+const ruleCondition2 = {
+    HttpErrorCodeReturnedEquals: 404,
+};
+const config = new WebsiteConfigTester('index.html', 'error.html');
+config.addRoutingRule(ruleRedirect1, ruleCondition1);
+config.addRoutingRule(ruleRedirect2, ruleCondition2);
+
+describe('GET bucket website', () => {
+    withV4(sigCfg => {
+        const bucketUtil = new BucketUtility('default', sigCfg);
+        const s3 = bucketUtil.s3;
+
+        afterEach(() => {
+            process.stdout.write('about to delete bucket\n');
+            return bucketUtil.deleteOne(bucketName)
+            .catch(err => {
+                if (err) {
+                    process.stdout.write('error in afterEach', err);
+                    throw err;
+                }
+            });
+        });
+
+        describe('with existing bucket configuration', () => {
+            before(done => {
```

Comment:
Same here, if you leverage promises
```js
    before(() =>
        s3.createBucket({ Bucket: bucketName })
        .then(() => s3.putBucketWebsite({
            Bucket: bucketName,
            WebsiteConfiguration: config,
        }))
```
You would have to make sure you return the promise always, otherwise mocha weirdly passes the tests even when there are failures.